docsource/paging.rst

```diff
@@ -0,0 +1,158 @@
+.. _paging:
+
+Paging workspace objects
+========================
+
+Workspaces can contain many objects, and the ``list_objects`` command returns a maximum of
+10000 ``object_info`` data structures (referred to as objects from now on) at a time. This
+section of the documentation discusses how to handle the case where access to large amounts
+of data is required.
+
+If it is reasonable to expect that a ``list_objects`` command will return no more than 10000
+objects then paging doesn't need to be considered. For example if the command only encompasses
+one workspace, only the most recent version of each object is required, and the ``max_objid``
+field in the workspace information returned by various API methods (for example
+``get_workspace_info``) is no more than 10000, then a single ``list_objects`` command can return
+all the data requested. See :ref:`workspaces` for more information about getting information
+about workspaces and :ref:`listobjects` for information about the ``list_objects`` command.
+
+There are currently two ways to page through workspace data - via object ID limits, and via
+reference limit.
+
+Parameters to avoid when paging
```

Comment:
It might make more sense to describe the methods of paging first, and then talk about paging params to avoid.

Reply:
Here I respectfully disagree. If I describe how to perform the technique and *then* describe the pitfalls, no one is going to read the pitfalls section and they'll have enough rope to hang themselves. IMO tell the user now not to screw up the technique first, and then explain the tecnique